source/options.ts

```diff
@@ -0,0 +1,52 @@
+import {Queue} from './queue';
+
+// tslint:disable:no-redundant-jsdoc
+
+export interface QueueAddOptions {
+	[key: string]: unknown;
+}
+
+export interface Options<Q extends Queue<O>, O extends QueueAddOptions> {
```

Comment:
Can you use more descriptive names for the type variables? `Q` is not very readable.